Assets/_Game/Scripts/Inventory/ItemDatabaseObject.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "New Item Database", menuName = "DLS/Item Database")]
+public class ItemDatabaseObject : ScriptableObject, ISerializationCallbackReceiver
+{
+    public ItemObject[] items;
+
+    public Dictionary<ItemObject, string> GetName = new Dictionary<ItemObject, string>();
+    public Dictionary<string, ItemObject> GetItem = new Dictionary<string, ItemObject>();

```

Comment:
This seems like both a waste of space (storing an entire copy of the itemobject) and poor naming (dictionary as what you expect a get method name). Are you doing this just to have a O(1) lookup for items? If that's the case, it'd be better to wrap these, store the index instead of a copy of the ItemObject, and push back what you want instead.